What's new article (#44643)
dotnet/docs: docs/whats-new/dotnet-docs-mod1.md

- [SYSLIB0060: Rfc2898DeriveBytes constructors are obsolete](../fundamentals/syslib-diagnostics/syslib0060.md)

### Updated articles

- [Creating metrics](../core/diagnostics/metrics-instrumentation.md) - [diagnostics] Add InstrumentAdvice details to instrumentation doc
- [dotnet-coverage code coverage utility](../core/additional-tools/dotnet-coverage.md) - Update dotnet-coverage docs. Adding uninstrument command
- [Microsoft.Testing.Platform and VSTest comparison](../core/testing/unit-testing-platform-vs-vstest.md) - Add first level info for migration off of VSTest
- [Tutorial: Containerize a .NET app](../core/docker/build-container.md) - Address issues related to .NET containers

## C# language

### New articles

- [Custom attributes that generate flags or options in the Intermediate Language (IL) output](../csharp/language-reference/attributes/pseudo-attributes.md)

## Visual Basic language

### Updated articles

- [Generic types in Visual Basic (Visual Basic)](../visual-basic/programming-guide/language-features/data-types/generic-types.md) - Add VB new features

## AI in .NET

### New articles

> - [Create a minimal AI assistant using .NET](../ai/quickstarts/quickstart-assistants.md)

Reviewer: Ahh, here it is @BillWagner. Could we make this the first section in the what's new?

Reviewer 2: Yes! 

It's already in the algorithm.  All we need to do is change the order in the `.whatsnew.json` file. 

I'll make a PR tomorrow (after merging this) to update the config and move the existing sections. That way, future diffs will be easier to understand.

I'll suggest putting the AI section after the security and breaking changes, but before everything else. Do you agree, or should AI be above those as well?

Reviewer: That sounds good, thanks.

Reviewer 2: Implemented in #44677

- [SYSLIB0060: Rfc2898DeriveBytes constructors are obsolete](../fundamentals/syslib-diagnostics/syslib0060.md)

### Updated articles

- [Creating metrics](../core/diagnostics/metrics-instrumentation.md) - [diagnostics] Add InstrumentAdvice details to instrumentation doc
- [dotnet-coverage code coverage utility](../core/additional-tools/dotnet-coverage.md) - Update dotnet-coverage docs. Adding uninstrument command
- [Microsoft.Testing.Platform and VSTest comparison](../core/testing/unit-testing-platform-vs-vstest.md) - Add first level info for migration off of VSTest
- [Tutorial: Containerize a .NET app](../core/docker/build-container.md) - Address issues related to .NET containers

## C# language

### New articles

- [Custom attributes that generate flags or options in the Intermediate Language (IL) output](../csharp/language-reference/attributes/pseudo-attributes.md)

## Visual Basic language

### Updated articles

- [Generic types in Visual Basic (Visual Basic)](../visual-basic/programming-guide/language-features/data-types/generic-types.md) - Add VB new features

> - [Troubleshooting Interoperability (Visual Basic)](../visual-basic/programming-guide/com-interop/troubleshooting-interoperability.md) - Fix up missing periods (.NET Framework docs)

Reviewer: ```suggestion
```

## AI in .NET

### New articles

- [Create a minimal AI assistant using .NET](../ai/quickstarts/quickstart-assistants.md)

## Azure SDK for .NET

### New articles

> - [ZZZ - Title not found in: #44284](../azure/sdk/authentication/authentication-best-practices.md)

Reviewer 2: ```suggestion
- [Authentication best practices with the Azure Identity library for .NET](../azure/sdk/authentication/best-practices.md)
```

## Microsoft Orleans

### New articles

- [Grain directory implementation](../orleans/implementation/grain-directory.md)

### Updated articles

- [Background Services and Startup Tasks](../orleans/host/configuration-guide/startup-tasks.md) - [Orleans] Update startup tasks guidance
- [Cluster management in Orleans](../orleans/implementation/cluster-management.md) - [Orleans] Update clustering implementation docs

## .NET Framework

### New articles

- [.NET Framework release notes](../framework/release-notes/release-notes.md)
- [January 2025 cumulative update](../framework/release-notes/2025/01-14-january-cumulative-update.md)

> 
> ### Updated articles
> 
> - [Security Concerns and Useful Tips for Tracing](../framework/wcf/diagnostics/tracing/security-concerns-and-useful-tips-for-tracing.md) - Fix up missing periods (.NET Framework docs)
> - [Security Concerns for Message Logging](../framework/wcf/diagnostics/security-concerns-for-message-logging.md) - Fix up missing periods (.NET Framework docs)

Reviewer: ```suggestion
```
